src/axom/core/Array.hpp

```diff
@@ -1004,6 +1097,14 @@ inline void Array<T, DIM, SPACE>::updateNumElements(IndexType new_num_elements)
 {
   assert(new_num_elements >= 0);
   assert(new_num_elements <= m_capacity);
+  if(m_default_construct)
+  {
+    detail::initializeInPlace(m_data,
```

Comment:
It seems `std::vector<T>` instantiated with a non-default-constructible type doesn't actually support resizing or construction with initial sizes (i.e. you have to reserve capacity and/or push_back): https://godbolt.org/z/T1zYhq54h

We could probably static_assert on resizes and construction-with-size in these cases. (That can probably be a future discussion though)

Reply:
Should we create an issue to track this?
(or add it to #633 ?)